cmd/sdkch/main.go

```diff
@@ -0,0 +1,277 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+const (
+	entriesDirName = ".pending"
+)
+
+var (
+	progName string
+
+	entriesDir string
+
+	// sections name-title map
+	sections = map[string]string{
+		"breaking":     "Breaking Changes",
+		"features":     "New features",
+		"improvements": "Improvements",
+		"bugfixes":     "Bugfixes",
+	}
+	// stanzas name-title map
+	stanzas = map[string]string{
+		"gaia":       "Gaia",
+		"gaiacli":    "Gaia CLI",
+		"gaiarest":   "Gaia REST API",
+		"sdk":        "SDK",
+		"tendermint": "Tendermint",
+	}
+)
+
+func init() {
+	progName = filepath.Base(os.Args[0])
+	cwd, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err)
+	}
+	flag.StringVar(&entriesDir, "d", filepath.Join(cwd, entriesDirName), "entry files directory")
+	flag.Usage = printUsage
+}
+
+func main() {
+	log.SetFlags(0)
+	log.SetPrefix(fmt.Sprintf("%s: ", filepath.Base(progName)))
+	flag.Parse()
+
+	if flag.NArg() < 1 {
+		errInsufficientArgs()
+	}
+
+	cmd := flag.Arg(0)
+	switch cmd {
+
+	case "init":
+		initChangesDir()
+
+	case "add":
+		if flag.NArg() < 3 {
+			errInsufficientArgs()
+		}
+		if flag.NArg() > 4 {
+			errTooManyArgs()
+		}
+		sectionDir, stanzaDir := flag.Arg(1), flag.Arg(2)
+		validateSectionStanzaDirs(sectionDir, stanzaDir)
+		if flag.NArg() == 4 {
+			addSinglelineEntryFile(sectionDir, stanzaDir, strings.TrimSpace(flag.Arg(3)))
+			return
+		}
+		addEntryFile(sectionDir, stanzaDir)
+
+	case "generate":
+		version := "UNRELEASED"
+		if flag.NArg() > 1 {
+			version = strings.Join(flag.Args()[1:], " ")
+		}
+		generateChangelog(version)
```

Comment:
After the generate command is run, all the files are left in tact, which I think is fine if for instance the releaser messed up and wanted to re-generate the changelog information. However this indicates to me that we should also have an easy command to delete the pending changelog contents such as `sdkch clear`

Reply:
It's now implemented as a flag: `-prune`